packages/flutter/lib/src/semantics/semantics.dart

```diff
@@ -101,6 +101,70 @@ final int _kUnblockedUserActions =
     SemanticsAction.didGainAccessibilityFocus.index |
     SemanticsAction.didLoseAccessibilityFocus.index;
 
+/// Function signature for checks in [DebugSemanticsRoleChecks.kChecks].
+///
+/// The check is run against any `node` that is sent to the platform.
+///
+/// To access the flags and properties, one should call the
+/// [SemanticsNode.getSemanticsData].
+@visibleForTesting
+typedef DebugSemanticsRoleCheck = FlutterError? Function(SemanticsNode node);
+
+/// A static class to conduct semantics role checks.
+///
+/// When adding a new [SemanticsRole], one must also add a corresponding check
+/// to [kChecks].
+@visibleForTesting
+class DebugSemanticsRoleChecks {
+  DebugSemanticsRoleChecks._();
```

Comment:
Drive-by comment: this can be a `final class` ? Then `DebugSemanticsRoleChecks._();` is not needed.

Reply:
final class can still be instantiated I think? I use abstract instead

Reply:
Yeah, you're right, I had to double check the class modifiers reference :) 
But yeah, we should be able to eliminate the constructor anyway